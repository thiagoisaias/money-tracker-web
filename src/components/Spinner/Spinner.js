import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Container = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const StyledSpinner = styled.svg`
  animation: rotate 2s linear infinite;
  width: ${props => props.width || "24px"};
  height: ${props => props.height || "24px"};
  height: 80px;

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }

  @keyframes dash {
    0% {
      stroke-dasharray: 1, 150;
      stroke-dashoffset: 0;
    }
    50% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -35;
    }
    100% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -125;
    }
  }
`;

const StyledCircle = styled.circle`
  stroke: #777;
  stroke-linecap: round;
  stroke-width: 1.725;
  animation: dash 1.5s ease-in-out infinite;
`;

const Spinner = props => {
  const { width, height } = props;
  return (
    <Container>
      <StyledSpinner width={width} height={height} viewBox="0 0 50 50">
        <StyledCircle cx="25" cy="25" r="15" fill="none" />
      </StyledSpinner>
    </Container>
  );
};

Spinner.propTypes = {
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired
};

export default Spinner;
