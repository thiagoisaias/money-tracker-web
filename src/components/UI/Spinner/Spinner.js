import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Container = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const StyledSpinner = styled.svg`
  animation: rotate 2s linear infinite;
  width: ${props => props.width || "24px"};
  height: ${props => props.height || "24px"};

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }

  @keyframes dash {
    0% {
      stroke-dasharray: 1, 150;
      stroke-dashoffset: 0;
    }
    50% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -35;
    }
    100% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -125;
    }
  }
`;

const StyledCircle = styled.circle`
  stroke: #999;
  stroke-linecap: round;
  stroke-width: 1.725;
  animation: dash 1.5s ease-in-out infinite;
`;

const Spinner = props => {
  const { width, height } = props;
  return (
    <Container>
      <StyledSpinner width={width} height={height} viewBox={`0 0 ${width*2} ${height*2}`}>
        <StyledCircle cx={width} cy={height} r={width / 2} fill="none" />
      </StyledSpinner>
    </Container>
  );
};

Spinner.propTypes = {
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired
};

export default Spinner;
