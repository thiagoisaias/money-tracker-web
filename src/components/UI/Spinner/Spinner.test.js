import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Spinner from "./Spinner";

describe("Spinner", () => {
  const width = 24;
  const height = 24;
  const component = shallow(<Spinner width={width} height={height} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
