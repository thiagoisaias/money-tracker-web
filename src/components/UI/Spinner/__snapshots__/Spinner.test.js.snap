// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders properly 1`] = `
<styled.div>
  <styled.svg
    height={24}
    viewBox="0 0 48 48"
    width={24}
  >
    <styled.circle
      cx={24}
      cy={24}
      fill="none"
      r={12}
    />
  </styled.svg>
</styled.div>
`;
